include/nlohmann/detail/output/binary_writer.hpp

```diff
@@ -1531,23 +1597,26 @@ class binary_writer
     /*
     @brief write a number to output input
     @param[in] n number of type @a NumberType
+    @param[in] use_bjdata  whether write in BJData format, default is false
     @tparam NumberType the type of the number
     @tparam OutputIsLittleEndian Set to true if output data is
                                  required to be little endian
 
     @note This function needs to respect the system's endianness, because bytes
           in CBOR, MessagePack, and UBJSON are stored in network order (big
           endian) and therefore need reordering on little endian systems.
+          On the other hand, BSON and BJData use little endian and should reorder
+          on big endian systems.
     */
     template<typename NumberType, bool OutputIsLittleEndian = false>
-    void write_number(const NumberType n)
+    void write_number(const NumberType n, const bool use_bjdata = false)
```

Comment:
As above with function parameter vs template parameter.

Reply:
I thought about this, and tried, but it appears that
```
write_number<NumberType, use_bjdata>(n);
```
won't work because `use_bjdata` is not a constant expression. I had to do
```
if(use_bjdata)
    write_number<NumberType, true>(n);
else
    write_number<NumberType, false>(n);
```

for all `write_number` calls in UBJSON/BJData section, which seems a bit too verbose, so I decided to use a parallel condition to make the implementation shorter. Is there a more elegant way to avoid the above branch?

Reply:
I was thinking of going the other way, with removing the template parameter and always using the runtime parameter.

Reply:
I agree that merging the two options using a single param is cleaner. This is done in the below commit

https://github.com/nlohmann/json/pull/3336/commits/5c4ba261eb4db6430faab76e5acc66996f7fdd27